R2Northstar/NorthstarLauncher
# Remove uses of Autohook from `serverauthentication.cpp`

### Code review:
- The original function is prefixed with `o_p` so any times where the old AUTOHOOK was calling the original function it should now be calling that instead.
- The hook function is prefixed with `h_` so any times where we would be wanting to call the hooked function from other functions should now be calling that instead
- Something to note is that we get a pointer to `CBaseClient__Disconnect` in another file as well, hence the weird name on the original hook. Potentially any calls we were making to `CBaseClient__Disconnect` instead of `_CBaseClient__Disconnect` are bypassing the hook? Something worth investigating at some point, shouldn't affect this PR though.

### Testing:
- Check logs to make sure that all of the changed hooks are being created properly
- Make sure that players can still auth, join and leave a server with their persistence being written etc.
